### DISCUSSION TITLE
MacOS

### DISCUSSION THREAD

**Discussion by quartz-dk** [FormerLurker/ArcWelderLib#40] [Q&A]
Hi

I am absolutely thrilled by the aspects of ArcWelder (although is already had an suspicion, that Arc where either not implemented or supported in printer firmware) - as this may make things so much nicer :-)

My issue here is, that I only have MacOS and I haven't found a way, to make any of the software run on the Mac (I am unfortunately no code genius), so if Mac binaries are available or  guides to compile a version are located at another location, I would be very happy if pointed to that site.

The is only two files in the (old) repository: ArcWeldeR.py and ArcWelder and MacOS' Python seems not to like them :-(

Any help will be appreciated :-)

**Comment by FormerLurker** [comment]
See [the latest release](https://github.com/FormerLurker/ArcWelderLib/releases).  There is a MacOS build there provided by @FieldOfView.  Let me know if it works for you.

**Reply by quartz-dk** [reply]
Hi - it works. Thank you :-)

I can see, that I need to do a little more reading and tweaking, but it works and I see a general compression ratio of about 60%. "The "smoothing" was not as big as  expected, but that's where I need to do the reading and tweaking I think :-D

Thank you very much and have a really splendid weekend.

**Comment by atas80** [comment]
Hello 

How to use the plugin in Prusa for MAC OS

This text from terminal - iMac-27-new:bin imac27_2014$ /Users/imac27_2014/Desktop/macOS/bin/ArcWelder -z ToPrint.gcode
2021-05-26 17:01:09.002 - arc_welder.gcode_conversion - INFO - Source and target path are the same.  The source file will be overwritten.  Temporary file path: 719a288e-3d85-c237-7184-f1d585b68792.tmp
2021-05-26 17:01:09.002 - arc_welder.gcode_conversion - INFO - Processing Gcode
	Source File Path             : ToPrint.gcode
	Target File Path (overwrite) : 719a288e-3d85-c237-7184-f1d585b68792.tmp
	Temporary File Path          : 719a288e-3d85-c237-7184-f1d585b68792.tmp
	Resolution                   : 0.0500mm (+-0.02500mm)
	Path Tolerance               : 5.000%
	Maximum Arc Radius           : 1000000mm
	Min Arc Segments             : 0
	MM Per Arc Segment           : 1000000.000
	Allow Z-Axis Changes         : True
	G90/G91 Influences Extruder  : False
	Log Level                    : INFO
	Hide Progress Updates        : False
Segmentation fault: 11


PrusaSlicer - error: 11
